server/src/test/scala/org/http4s/server/middleware/ErrorHandlingSuite.scala

```diff
@@ -54,4 +55,47 @@ class ErrorHandlingSuite extends Http4sSuite {
       .map(_.status)
       .assertEquals(Status.BadRequest)
   }
+
+  sealed trait ErrorCode extends Throwable
+  object ErrorCode {
+    case object CouldNotParseRequest extends ErrorCode
+  }
+
+  test("Handle errors with a custom error handler") {
+    ErrorHandling(
+      routes(ErrorCode.CouldNotParseRequest),
+      (req: Request[IO]) => {
+        case ErrorCode.CouldNotParseRequest => OptionT.liftF(BadRequest())
+      }
+    ).orNotFound
+      .apply(request)
+      .map(_.status)
+      .assertEquals(Status.BadRequest)
+  }
+
+  test("Handle errors with a custom error handler via httpRoutes") {
+    ErrorHandling.httpRoutes(
+      routes(ErrorCode.CouldNotParseRequest),
+      (req: Request[IO]) => {
```

Comment:
A type hint needs to be supplied, the compiler seems not to be able to infer that it is `Request[IO]`

Reply:
You should be able to do what you want by having `errorHandler` in another argument group